.generator/templates/model/model.mustache

```diff
@@ -65,11 +67,20 @@ export class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{
       if ([{{#enumVars}}{{{value}}}, {{/enumVars}}undefined].includes(data.{{{baseName}}})) {
           res.{{name}} = data.{{{baseName}}};
       } else {
-          throw TypeError(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          console.error(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          throw new ParseException("enum", `invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
       }
         {{/allowableValues}}
         {{^allowableValues}}
-      res.{{{name}}} = ObjectSerializer.deserialize(data.{{{baseName}}}, "{{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}", "{{dataFormat}}")
+        try {
+          res.{{{name}}} = ObjectSerializer.deserialize(data.{{{baseName}}}, "{{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}", "{{dataFormat}}")
+        } catch(error) {
+            if (res?.unparsedObject === undefined) {
+              res.unparsedObject = {};
+            }
+            res.unparsedObject["{{{name}}}"] = data.{{{baseName}}};
+        }
```

Comment:
At this point, deserialise errors for enums and oneOf should already be taken care of. Any exception here should be legit and not caught.
We don't wan't to set `unparsedObject` for a single property anyway. It's all or nothing.